src/main/java/org/example/jvspringbootfirstbook/service/order/OrderServiceImpl.java

```diff
@@ -0,0 +1,164 @@
+        return orderRepository.save(order);
+    }
+
+    private Set<OrderItem> calculatePriceForSet(Set<OrderItem> orderItemSet) {
+        return orderItemSet.stream()
+                .map(this::setPrice)
+                .collect(Collectors.toSet());
+    }
+
+    private OrderItem setPrice(OrderItem orderItem) {
+        orderItem.setPrice(
+                orderItem.getBook().getPrice().multiply(
+                        BigDecimal.valueOf(orderItem.getQuantity())
+                )
+        );
+        return orderItem;
+    }
+
+    private Set<OrderItem> getOrderItemSet(ShoppingCart cart, Order emptyOrder) {
+        Set<CartItem> cartItems = cart.getCartItems();
+        return cartItems.stream()
+                .map(orderItemMapper::fromCarttoOrderItem)
+                .map(orderItem -> setOrderToItem(orderItem, emptyOrder))
+                .collect(Collectors.toSet());
+    }
+
+    private OrderItem setOrderToItem(OrderItem orderItem, Order emptyOrder) {
+        orderItem.setOrder(emptyOrder);
+        return orderItem;
+    }
+
+    @Override
+    public List<OrderDto> getHistory(User user, Pageable pageable) {
+        return orderRepository.findAllByUser(user, pageable).stream()
+                .map(orderMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public List<OrderItemDto> getFromOrder(Long orderId) {
+        Order order = orderRepository.findById(orderId)
+                .orElseThrow(
+                        () -> new EntityNotFoundException(
+                                "We don't have order with this id: " + orderId)
+                );
+        return order.getOrderItems().stream()
+                .map(orderItemMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public OrderItemDto getFromOrder(Long orderId, Long id) {
+        Order order = orderRepository.findById(orderId)
+                .orElseThrow(
+                        () -> new EntityNotFoundException(
+                                "We don't have order with this id: " + orderId)
+                );
+        Set<OrderItem> orderItems = order.getOrderItems();
+        OrderItem orderItem = orderItems.stream()
+                .filter(item -> item.getId().equals(id))
+                .findFirst()
+                .orElseThrow(
+                        () -> new EntityNotFoundException(
+                                "We can't find order Item with this id"
+                        )
+                );
+        return orderItemMapper.toDto(orderItem);
+    }
+
+    @Override
+    public OrderDto updateStatus(Long id) {
+        if (!orderRepository.findById(id).isPresent()) {
+            throw new EntityNotFoundException("We don't have a order");
+        }
+        Order order = orderRepository.findById(id)
+                .orElseThrow(
+                        () -> new EntityNotFoundException("We don't have a order")
+                );
+        if (order.getStatus().equals(Status.PENDING)) {
```

Comment:
You could simplify the ifs using switch statement:
```
switch (order.getStatus()) {
            case PENDING:
                order.setStatus(Status.DELIVERED);
                break;
            case DELIVERED:
                order.setStatus(Status.COMPLETED);
                break;
            default:
                order.setDeleted(true);
                break;
        }
        return getDto(order);
```
If you want to stick to ifs:
- make order.getStatus a variable
- flip equals so that Status.XXX will be first - in that way you avoid NPE